### PR TITLE
Refresh token tests

### LINKED ISSUE #205: OAuth endpoint should respond to Refresh grant type
It's not in the spec, so check out stormpath/stormpath-framework-spec/issues/40

### PR DESCRIPTION
**Pull request by edjiang**
Fixes #205

### CODE REVIEW

File: src/test/groovy/com/stormpath/tck/oauth2/Oauth2IT.groovy [hunk 1]
```diff
@@ -203,4 +204,68 @@ class Oauth2IT extends AbstractIT {
...
         assertTrue(JwtUtils.extractJwtClaim(accessToken, "sub") == this.accountHref)
     }
+
+    /** Refresh grant flow
+     * @see <a href="https://github.com/stormpath/stormpath-framework-tck/issues/205">#205</a>
+     */
+    @Test
+    public void refreshGrantTypeWorksWithValidToken() throws Exception {
+        Response passwordGrantResponse =
+            given()
```
Comment (nbarbettini) [inline src/test/groovy/com/stormpath/tck/oauth2/Oauth2IT.groovy:214]: Probably should add `.accept(ContentType.JSON)` here

File: src/test/groovy/com/stormpath/tck/oauth2/Oauth2IT.groovy [hunk 2]
```diff
@@ -203,4 +204,68 @@ class Oauth2IT extends AbstractIT {
...
+                .body("refresh_token", not(isEmptyOrNullString()))
+                .body("token_type", is("Bearer"))
+            .extract()
+                .response()
+
+        String accessToken = passwordGrantResponse.path("access_token")
+        String refreshToken = passwordGrantResponse.path("refresh_token")
+
+        String newAccessToken =
+            given()
```
Comment (nbarbettini) [inline src/test/groovy/com/stormpath/tck/oauth2/Oauth2IT.groovy:234]: Ditto on Accept

File: src/test/groovy/com/stormpath/tck/oauth2/Oauth2IT.groovy [hunk 3]
```diff
@@ -203,4 +204,68 @@ class Oauth2IT extends AbstractIT {
...
+
+    /** Refresh grant flow should fail without valid refresh token
+     * @see <a href="https://github.com/stormpath/stormpath-framework-tck/issues/205">#205</a>
+     */
+    @Test
+    public void refreshGrantTypeFailsWithInvalidRefreshToken() throws Exception {
+        String refreshToken = "GARBAGE"
+
+        String newAccessToken =
+            given()
```
Comment (nbarbettini) [inline src/test/groovy/com/stormpath/tck/oauth2/Oauth2IT.groovy:261]: Accept header

**Comment by nbarbettini** [comment]
Other than specifying the `Accept` header in the places I noted, LGTM :+1:

**Comment by edjiang** [comment]
@nbarbettini, I'm going to disagree on the accept header. 

If you look at the OAuth spec (for password grant type, for instance), https://tools.ietf.org/html/rfc6749#section-4.3.2, they show a POST without an accept header working properly. The content type is always going to be application/json whether or not we accept it, since we should be able to deal with that response. 

If I connect to https://stormpath.com/images/navbar/navbar-logo.png

```
GET /images/navbar/navbar-logo.png HTTP/1.1
Accept: text/html
```

Our webserver is not going to respond with `406 Not Acceptable`, it's going to respond with the image. Since that's what the resource is. 

According to the HTTP protocol:

> Note: HTTP/1.1 servers are allowed to return responses which are not acceptable according to the accept headers sent in the request. In some cases, this may even be preferable to sending a 406 response. User agents are encouraged to inspect the headers of an incoming response to determine if it is acceptable.

I would say our servers should be liberal in what they accept, and requests without an `Accept` header should be treated just fine.

**Comment by nbarbettini** [comment]
@edjiang That logic makes sense. I agree :+1:

**Pull request merged**